AWSClientRuntime/Sources/Endpoints/AWSEndpointsResolvedEndpoint.swift

```diff
@@ -6,29 +6,49 @@ import AwsCommonRuntimeKit
 /// Wrapper for CRTAWSEndpointResolvedEndpoint
 public class AWSEndpointsResolvedEndpoint {
     
-    let crtResolvedEndpoint: CRTAWSEndpointResolvedEndpoint
+    let crtResolvedEndpoint: AwsCommonRuntimeKit.ResolvedEndpoint
     
-    init(crtResolvedEndpoint: CRTAWSEndpointResolvedEndpoint) {
+    init(crtResolvedEndpoint: AwsCommonRuntimeKit.ResolvedEndpoint) {
         self.crtResolvedEndpoint = crtResolvedEndpoint
     }
     
     public func getType() -> AWSEndpointsResolvedEndpointType {
-        AWSEndpointsResolvedEndpointType(crtType: crtResolvedEndpoint.getType())
+        AWSEndpointsResolvedEndpointType(crtType: crtResolvedEndpoint)
     }
     
-    public func getError() throws -> String? {
-        try crtResolvedEndpoint.getError()
+    public func getError() -> String? {
```

Comment:
this would require changes in endpoint resolver. you would have to include those changes in this PR to pass checks.